src/org/parosproxy/paros/extension/history/ProxyListenerLog.java

```diff
@@ -82,15 +82,17 @@ public boolean onHttpRequestSend(HttpMessage msg) {
 //	        return;
 //	    }
 	    
-	    HttpMessage existingMsg = model.getSession().getSiteTree().pollPath(msg);
-
-	    // check if a msg of the same type exist
-	    if (existingMsg != null && !existingMsg.getResponseHeader().isEmpty()) {
-	        if (HttpStatusCode.isSuccess(existingMsg.getResponseHeader().getStatusCode())) {
-	            // exist, no modification necessary
-	            return true;
-	        }
-	    }
+	    if (!Constant.isLowMemoryOptionSet()) {

```

Comment:
If this is to just address the NPE it should be used the method `SessionStructure.find` to get the `HttpMessage` (in the following statement), instead of skip the check entirely.